### ISSUE TITLE
TLS certificate validation

### ISSUE THREAD

**Issue by zoni** [karanlyons/chump#7]
85873ed961c949244a8dc4544d2e288b8e9063d6 removes the dependency on requests by replacing it with Python's built-in urllib module, however unlike requests, Python stdlib modules like urllib do not perform validation of TLS certificates by default ([yet](http://legacy.python.org/dev/peps/pep-0476/)).

From glimpsing at the code quickly, I could not find anything that ensures certificates are validated, so it seems to be possible to perform MITM attacks against pushover.net and have notifications intercepted with the current code.

**Comment by karanlyons** [comment]
Aw *crap*.

Now I remember why I was still keeping requests around.

I don’t think it’s *as* bad as you say (I’m very tired right now, but I’m pretty sure I tested this and was getting proper errors if I tried to MITM myself, but this was only testing with the latest versions of Python 2 and 3, both of which now validate by default), but that’s definitely bad.

I think I can just generate the proper `SSLContext` from ssl in the stdlib (since I’m only supporting >=2.6), but the flags for disabling SSL2/3 didn’t land until 2.7.9. So it’s not clear to me how to do this:

````python
import ssl

ssl_context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
ssl_context.options |= ssl.OP_NO_SSLv2 | ssl.OP_NO_SSLv3
ssl_context.verify_mode = ssl.CERT_REQUIRED
ssl_context.load_default_certs(ssl.Purpose.SERVER_AUTH)
````

In Python < 2.7.9. Some of those constants can just be redefined as they get passed to OpenSSL anyway, but some of those functions/attributes just don’t exist prior to 2.7.9.

It seems the best course of action is to do the best that we can with what is provided in the environment, with the best case scenario being something like the above: using the machine’s CA bundle and possibly checking CRLs for good measure. Next up would be certifi/requests (as they don’t use the machine’s certs), and if that failed as well there'd no verification at all and a big warning on first import.

It’s also possible that I’m wrong to prefer the machine’s bundle over certifi’s. And obviously the *easiest* solution would be to just require requests again, but that seems silly given how little of it chump actually needs.

Thoughts?
